File: backend/app/schemas/document.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime 

class DocumentBase(BaseModel):
    filename: str 
    file_path: str 
    
class DocumentCreate(DocumentBase):
    last_modified: Optional[float] = None 
    last_indexed: Optional[datetime] = None 
    
class ScanPendingFileResponse(BaseModel):
    files: List[str]

class SearchRequest(BaseModel):
    query: str
    n_results: int = 10
    unique_results: bool = True

class SearchResult(DocumentBase):
    excerpt: str 
    distance: float 
    
class SearchResponse(BaseModel):
    results: list[SearchResult]

File: frontend/src/types/document.ts
export interface PendingFilesResponse {
  files: string[];
}

export interface IndexFileRequest {
  file_path: string;
}

export interface SearchRequest {
  query: string;
  n_results?: number;
}

export interface SearchResult {
  file_path: string;
  filename: string;
  excerpt: string;
  distance: number;
}

export interface SearchResponse {
  results: SearchResult[];
}
